engine/runtime/src/main/java/org/enso/interpreter/runtime/Module.java

```diff
@@ -383,11 +383,17 @@ public Source getSource() throws IOException {
    */
   @TruffleBoundary
   public final SourceSection createSection(int sourceStartIndex, int sourceLength) {
-    var src = sources.source();
+    Source src;
+    try {
+      src = getSource();
+    } catch (IOException e) {
+      TruffleLogger logger = TruffleLogger.getLogger(LanguageInfo.ID, Module.class);
+      logger.log(Level.SEVERE, "Failed to retrieve sources of the module: " + e.getMessage(), e);
+      return null;
+    }
```

Comment:
The `if src == null` check can be removed if you move the rest of the code into the `try/catch` block.